simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileBlockStreamManager.java

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.generator;
+
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
+import com.hedera.block.simulator.config.types.GenerationMode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+/** The block as file block stream manager. */
+public class BlockAsFileBlockStreamManager implements BlockStreamManager {
+
+    private static final System.Logger LOGGER =
+            System.getLogger(BlockAsFileBlockStreamManager.class.getName());
+
+    final String rootFolder;
+
+    final List<Block> blockList = new ArrayList<>();
+
+    int currentBlockIndex = 0;
+    int currentBlockItemIndex = 0;
+    int lastGivenBlockNumber = 0;
+
+    /**
+     * Constructor for the block as file block stream manager.
+     *
+     * @param blockStreamConfig the block stream config
+     */
+    @Inject
+    public BlockAsFileBlockStreamManager(@NonNull BlockStreamConfig blockStreamConfig) {
+        this.rootFolder = blockStreamConfig.folderRootPath();
+        try {
+            this.loadBlocks();
+        } catch (IOException | ParseException | IllegalArgumentException e) {
+            LOGGER.log(System.Logger.Level.ERROR, "Error loading blocks", e);
+            throw new RuntimeException(e);
+        }
+
+        LOGGER.log(System.Logger.Level.INFO, "Loaded " + blockList.size() + " blocks into memory");
+    }
+
+    @Override
+    public GenerationMode getGenerationMode() {
+        return GenerationMode.DIR;
+    }
+
+    @Override
+    public BlockItem getNextBlockItem() {
+        BlockItem nextBlockItem =
+                blockList.get(currentBlockIndex).items().get(currentBlockItemIndex);
+        currentBlockItemIndex++;
+        if (currentBlockItemIndex >= blockList.get(currentBlockIndex).items().size()) {
+            currentBlockItemIndex = 0;
+            currentBlockIndex++;
+        }
+        return nextBlockItem;
+    }
+
+    @Override
+    public Block getNextBlock() {
+        Block nextBlock = blockList.get(currentBlockIndex);
+        currentBlockIndex++;
+        lastGivenBlockNumber++;
+        if (currentBlockIndex >= blockList.size()) {
+            currentBlockIndex = 0;
+        }
+        return nextBlock;
+    }
+
+    private void loadBlocks() throws IOException, ParseException {
+
+        Path rootPath = Path.of(rootFolder);
+
+        try (Stream<Path> blockFiles = Files.list(rootPath)) {
+
+            List<Path> sortedBlockFiles =
+                    blockFiles.sorted(Comparator.comparing(Path::getFileName)).toList();
```

Comment:
I think you can remove the sort functionality here and build the path based on a loop index. If we eventually need to stream a lot of block items, sorting will impact performance.

Reply:
Agree with @mattp-swirldslabs on this. But also on the project, as a whole, I think that we should try to avoid loops and operations like this on hot paths, as much as possible.